### DISCUSSION TITLE
Suggestion translate

### DISCUSSION THREAD

**Discussion by newst4rt** [lyricstify/lyricstify#49] [Ideas]
Sometimes the lyrics will be not translated. I think it is a problem of google,  too many requests will be blocked.

My suggestion is to switch from "google-translate-api-x" to the "translate" package. The advantage of the newer package is the choice to switch the engine. I belief that "libretranslate" it is the best choice to translate the lyrics because you can use it in offline mode.  As well you have the possibility to use much other online translator inside the package.

I hope that my suggestion find your ear.

**Comment by yusuftaufiq** [comment]
That's a good suggestion. I agree that it seems that in the last few days, the rate limit from Google has been changed and causing more failed translations.

Can you also give me the GitHub repository link to your suggested packages? I looked it up on GitHub and found a lot of results, I was wondering which right package you're referring to.

**Comment by newst4rt** [comment]
https://www.npmjs.com/package/translate

**Reply by yusuftaufiq** [reply]
I'll try to check it out with that package. I'm also considering caching lyrics and translations. So that the API request hits to the translation service can be greatly reduced and the loading time before displaying lyrics can be much faster.

**Reply by newst4rt** [reply]
Google is a problem. When you reduce the translation transaction, you will be able to hit more translation, but you have the risk that you will be blocked. One other solution is to buy API access, but I think nobody want to pay for translated lyrics. The best way is to give the user the choice whitch translation service he want to use. When you adding inside lyricstify a additional prefix function, which translation engine and api-key  be can used I am sure that you fix the problem for the future. Note, that you have to add for libretranslate a url function for selfhosting translation service.

**Reply by yusuftaufiq** [reply]
Yeah, I'm planning to use your suggested package and combine it with the new caching feature and maybe also use a proxy to automatically bypass the rate limit. Let's see how this can be implemented.

**Reply by newst4rt** [reply]
Great ! A proxy feature is interesting but cautious; 99% of all public proxys are blacklisted. This feature will only work, when you have a provider with non blacklisted proxys and for that you have to pay.

**Reply by yusuftaufiq** [reply]
Nice information, thanks for letting me know.
